core/unit_test/TestRangePolicyScan.hpp

```diff
@@ -0,0 +1,228 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <Kokkos_Core.hpp>
+#include <cstdio>
+
+// This test checks parallel_scan() calls which use RangePolicy.
+
+namespace {
+
+template <typename ValueType>
+struct TestRangePolicyScan {
```

Comment:
Per our discussion on Slack, I think `TestParallelScanRangePolicy` is more intuitive.  Also applies to file names.